src/test/java/com/gooddata/warehouse/WarehouseServiceIT.java

```diff
@@ -243,4 +244,54 @@ public void shouldFailToAddUserToWarehouse() throws Exception {
         gd.getWarehouseService().addUserToWarehouse(warehouse, new WarehouseUser("role", "profile", null)).get();
     }
 
+    @Test
+    public void shouldRemoveUserFromWarehouse() throws Exception {
+        onRequest()
+                .havingMethodEqualTo("DELETE")
+                .havingPathEqualTo(WAREHOUSE_USER_URI)
+                .respond()
+                .withBody(readFromResource(TASK_POLL))
+                .withStatus(202);
+        onRequest()
+                .havingMethodEqualTo("GET")
+                .havingPathEqualTo(pollingTask.getPollLink())
+                .respond()
+                .withStatus(202)
+                .thenRespond()
+                .withBody(readFromResource(REMOVE_USER_TASK_DONE))
+                .withStatus(201);
+
+        gd.getWarehouseService()
+          .removeUserFromWarehouse(new WarehouseUser("role", "profile", "login", Collections.singletonMap("self", WAREHOUSE_USER_URI)))
+          .get();
+    }
+
+    @Test(expectedExceptions = WarehouseUserNotFoundException.class)
+    public void shouldFailToFindUserForRemovalFromWarehouse() throws Exception {
+        onRequest()
+                .havingMethodEqualTo("DELETE")
+                .havingPathEqualTo(WAREHOUSE_USER_URI)
+                .respond()
+                .withStatus(404);
+
+        gd.getWarehouseService().removeUserFromWarehouse(new WarehouseUser("role", "profile", "login", Collections.singletonMap("self", WAREHOUSE_USER_URI)));
+    }
+
+    @Test(expectedExceptions = GoodDataException.class)
+    public void shouldFailWhenRemoveUserTaskFails() throws Exception {
+        onRequest()
+                .havingMethodEqualTo("DELETE")
+                .havingPathEqualTo(WAREHOUSE_USER_URI)
+                .respond()
+                .withBody(readFromResource(TASK_POLL))
+                .withStatus(202);
+        onRequest()
+                .havingMethodEqualTo("GET")
+                .havingPathEqualTo(pollingTask.getPollLink())
+                .respond()
+                .withStatus(409);
+
+        gd.getWarehouseService().addUserToWarehouse(warehouse, new WarehouseUser("role", "profile", null)).get();
```

Comment:
this test is named "shouldFailWhenRemoveUserTaskFails", but calls "addUserToWarehouse" method?